### DISCUSSION TITLE
How to build?

### DISCUSSION THREAD

**Discussion by Psycheros** [EDCD/EDDI#2286] [EDDI Standalone Q&A] [answered]
Admittingly I'm not a windows person but I have a computer science background and usually use Linux. Using the latest version of Visual Studio Community Edition I cannot build this project and I feel like I've expended every outlet for building this out of the box. Is there something I'm missing? Where should I start?

**Comment by Tkael** [comment, marked as answer]
I'm on Windows and I think that there are a number of dependencies which require Windows. But have you set EDDI as the ~default~ *Startup* project?

**Reply by Psycheros** [reply]
I'm not running linux for this project. I am on Windows trying to build the project with Visual Studio. I've not set it as my default project because I prefer having none and deliberately file > opening what I'm working on. In this case, I used Git but still. No, not my default project.  
[edit: Not Visual Basic. Visual Studio]

**Reply by Tkael** [reply]
Ah. I meant "Startup project" rather than "Default project".

**Reply by Tkael** [reply]
![image](https://user-images.githubusercontent.com/4382539/141515205-9abde763-343b-4548-b87c-8f180455a97a.png)

**Reply by Psycheros** [reply]
I'd like to avoid setting defaults if at all possible. I'm comfortably deliberately selecting the project the hard way. 
Specifically the error I get is that 'git --version' failed to run, and that git might not be installed correctly. I've installed and reset once before building and once before starting this post.
 
Building anyway the top error is " that C:\...\...\containing folder\bin\Debug\Utilities.dll' could not be found, There are 5 total errors like this. 
As I understand it .dll should be generated while building, so I'm at a loss.

**Reply by Tkael** [reply]
In that case, please review the build output log to determine whether any errors are being logged during the build process? Or is that the output that you are getting during the build process?

**Reply by Tkael** [reply]
You might need to [install Git](https://git-scm.com/downloads) if you haven't already.

**Reply by Tkael** [reply]
If Git is already installed and the build is still failing then the build may not be able to locate the file it needs to run `git --version`. In that case, check your `Environment Variables` and make sure that the system path contains a reference to `Program Files\Git\cmd`.

**Reply by Psycheros** [reply]
So to clarify. It is not possible to build from a download without more help from git, even though I received all the files?

**Reply by Tkael** [reply]
We've had a number of commanders generate their own builds. It's hard for me to pinpoint the precise problem from the limited information that is available to me, but it sounds like your build logs are telling you that you are missing something needed by one of the packages we use (specifically, https://www.nuget.org/packages/GitInfo/).

**Reply by Psycheros** [reply]
so I needed to add my git path to my systems environment variables despite not intending to use git inside visual studio. The project build was successful. 
Why is this project dependent on git?

**Reply by Tkael** [reply]
We send the git commit with the telemetry data we collect. The package we use to obtain the git commit relies on git being installed.
